File: nodes/get-first-n-messages/get-first-n-messages.process.py
from typing import Any, Dict, List, Union

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    try:
        messages = inputs.get('messages')
        count = inputs.get('count')

        if not isinstance(messages, list):
            raise Exception("Messages input must be an array")

        if not isinstance(count, int) or count < 0:
            raise Exception("Count must be a non-negative integer")

        # Get the first N messages
        first_messages = messages[:count] if count > 0 else []
        actual_count = len(first_messages)

        return {
            "messages": first_messages,
            "count": actual_count
        }

    except Exception as error:
        print(f'Get First N Messages error: {error}')
        raise Exception(f"Get First N Messages error: {str(error)}")

File: sdks/nodejs/nodes/get-first-n-messages/get-first-n-messages.process.js
export default async ({inputs, settings, config}) => {
    try {
        const messages = inputs.messages;
        const count = inputs.count;

        if (!Array.isArray(messages)) {
            throw new Error("Messages input must be an array");
        }

        if (typeof count !== 'number' || count < 0 || !Number.isInteger(count)) {
            throw new Error("Count must be a non-negative integer");
        }

        // Get the first N messages
        const firstMessages = messages.slice(0, count);
        const actualCount = firstMessages.length;

        return {
            messages: firstMessages,
            count: actualCount
        };

    } catch (error) {
        console.log('Get First N Messages error:', error);
        throw new Error(`Get First N Messages error: ${error.message}`);
    }
};
